$OpenBSD: patch-parser-specs_commands_spec,v 1.3 2013/01/30 22:19:43 dcoppa Exp $

From cf830f69a2467b79454b6916ccfbbff45abf5418 Mon Sep 17 00:00:00 2001
From: Steven Allen <steven@stebalien.com>
Date: Tue, 29 Jan 2013 08:05:23 +0000
Subject: Bugfix: Correctly parse `move ... workspace *_on_output`

The parse spec for `move ... workspace ...`  ordered next/prev before
next_on_output/prev_on_output causing the parser to match next/prev
before next_on_output/prev_on_output.

Ticket: http://bugs.i3wm.org/report/ticket/941

--- parser-specs/commands.spec.orig	Wed Dec 12 00:08:17 2012
+++ parser-specs/commands.spec	Wed Jan 30 22:08:08 2013
@@ -223,7 +223,7 @@ state RENAME_WORKSPACE_NEW_NAME:
       -> call cmd_rename_workspace($old_name, $new_name)
 
 # move <direction> [<pixels> [px]]
-# move [window|container] [to] workspace [<str>|next|prev|current]
+# move [window|container] [to] workspace [<str>|next|prev|next_on_output|prev_on_output|current]
 # move [window|container] [to] output <str>
 # move [window|container] [to] scratchpad
 # move workspace to [output] <str>
@@ -264,7 +264,7 @@ state MOVE_DIRECTION_PX:
 state MOVE_WORKSPACE:
   'to'
       -> MOVE_WORKSPACE_TO_OUTPUT
-  workspace = 'next', 'prev', 'next_on_output', 'prev_on_output', 'current'
+  workspace = 'next_on_output', 'prev_on_output', 'next', 'prev', 'current'
       -> call cmd_move_con_to_workspace($workspace)
   'back_and_forth'
       -> call cmd_move_con_to_workspace_back_and_forth()
